packages/react-native-codegen/package.json

```diff
@@ -50,7 +50,7 @@
     "invariant": "^2.2.4",
     "micromatch": "^4.0.4",
     "mkdirp": "^0.5.1",
-    "prettier": "2.8.8",
+    "prettier": "3.0.1",
```

Comment:
I'm afraid upgrading prettier to v3 may break codegen as[ `prettier.format()` has been changed to an async API since v3](https://prettier.io/blog/2023/07/05/3.0.0#change-public-apis-to-asynchronous-12574httpsgithubcomprettierprettierpull12574-12788httpsgithubcomprettierprettierpull12788-12790httpsgithubcomprettierprettierpull12790-13265httpsgithubcomprettierprettierpull13265-by-fiskerhttpsgithubcomfisker).

[This](https://github.com/facebook/react-native/pull/38545/files#diff-1cd3300bccfcba4718c069421e21a9f51ae4a0aa362bcd1a0ca6eaf37fa9b08d) offers one possible solution and can serve as a reference.